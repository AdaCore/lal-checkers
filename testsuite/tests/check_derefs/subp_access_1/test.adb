procedure Test is
   procedure P is
   begin
      null;
   end;

   F : access procedure := null;
   B : Boolean;
begin
   if B then
      F := P'Access;
   end if;

   F.all;
end Ex1;
