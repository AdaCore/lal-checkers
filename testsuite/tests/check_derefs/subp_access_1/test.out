{
  "always": [
    {
      "precise": false,
      "trace": [
        "assign0",
        "assign1",
        "assume0",
        "assume1",
        "assume2",
        "read0",
        "split_join0",
        "start0"
      ]
    }
  ],
  "le_t_eq_v": [
    {
      "precise": true,
      "trace": [
        "assign0",
        "assume1",
        "assume2",
        "read0",
        "split_join0",
        "start0"
      ]
    }
  ]
}